packages/client/src/pages/Library/components/EditForm/EditForm.jsx

```diff
@@ -1,7 +1,275 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { toast } from 'react-toastify';
 
-const EditForm = () => {
-  return <div>EditForm</div>;
+// api
+import api from 'util/api';
+
+// constants
+import {
+  ICON_TYPES,
+  BTN_TYPES,
+  EDIT_FORM_VIEWS,
+  API_ROUTES,
+} from 'util/constants';
+
+// components
+import Icon from 'components/Icon/Icon';
+import LoadingSpinner from 'components/LoadingSpinner/LoadingSpinner';
+import Input from 'components/Input/Input';
+import Button from 'components/Button/Button';
+
+// styles
+import styles from './EditForm.module.css';
+
+const EditForm = ({ onSuccess, viewType, initialState }) => {
+  const [data, setData] = useState(initialState);
+  const { artists, albums } = EDIT_FORM_VIEWS;
+
+  const handleInputChange = (e) => {
+    setData({
+      ...data,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleRemoveBandMember = (index) => {
+    setData({
+      ...data,
+      bandMembers: [
+        ...data.bandMembers.filter(
+          (member) => member !== data.bandMembers[index]
+        ),
+      ],
+    });
+  };
+
+  const renderBandMembers = () =>
+    data.bandMembers.map((bandMember, index) => (
+      // eslint-disable-next-line
+      <div key={index} className={styles.bandMemberWrapper}>
+        <p className={styles.bandMember}>{bandMember}</p>
+        <Button
+          isWrapper
+          onClick={() => handleRemoveBandMember(index)}
+          className={styles.iconBtnMinus}
+          type='button'
+        >
+          <Icon iconType={ICON_TYPES.cancel} className={styles.iconSmall} />
+        </Button>
+      </div>
+    ));
+
+  const handleAddBandMember = () => {
+    setData({
+      ...data,
+      bandMembers: [...data.bandMembers, data.newBandMember],
+      newBandMember: '',
+    });
+  };
+
+  const handleSubmitArtist = async (e) => {
+    e.preventDefault();
+
+    setData({
+      ...data,
+      isSubmitting: true,
+      errorMessage: null,
+    });
+
+    try {
+      const body = {
+        artistTitle: data.artistTitle,
+        bandMembers: data.bandMembers,
+        description: data.description,
+      };
+
+      const response = await api.put(
+        `${API_ROUTES.artists.update}/${data.artistId}`,
+        body
+      );
+
+      toast.success(response.data.msg);
+
+      onSuccess();
+    } catch (error) {
+      setData({
+        ...data,
+        isSubmitting: false,
+        errorMessage: error ? error.message || error.statusText : null,
+      });
+      // eslint-disable-next-line
+      console.log(error);

```

Comment:
в идеале нужно убрать либо как минимум так
```suggestion
      console.error(error);

```